karaf/deploy-and-run.sh

```diff
@@ -1,23 +1,41 @@
 #!/bin/sh
 
-DIR="/deployments"
+DIR="/deployments/"
 
-echo "Checking for *.tar.gz in $DIR"
-# there should be only one *.tar.gz in ${DIR}
-NUM_ARCHIVE_FILES=`ls -1 ${DIR} | grep "^.*.tar.gz$" | wc -l`
-if [ $NUM_ARCHIVE_FILES -ne 1 ]; then
-  echo "Missing or more than one assembly archive file *.tar.gz in ${DIR}"
-  exit 1
-fi
-KARAF_ASSEMBLY_ARCHIVE=`ls -1 ${DIR}/*.tar.gz`
+echo "Checking for *.zip in $DIR"
+
+if [ `ls -1 ${DIR} | grep "^.*.zip$" | wc -l` -eq 1 ]; then
```

Comment:
* -1 is not necessary since the command has no tty anyway (so its always one entry per line
* Regexp is better "^.*\.zip$" in order to match the dot literally (and not as metachar). Othewies "blazip" would match as well.